src/complex/Utilities/Parsing/HDF5/H5ObjectReader.cpp

```diff
@@ -43,9 +49,26 @@ H5::IdType H5::ObjectReader::getParentId() const
   return m_ParentId;
 }
 
+size_t H5::ObjectReader::getObjectId() const
```

Comment:
```suggestion
haddr_t H5::ObjectReader::getObjectId() const
```
size_t changes it's underlying type across systems. Stick to the HDF5 type or create an alias that does the same. This also protects against HDF5 changing the type alias. I vaguely remember happening that happening with herr_t or hid_t several years ago when we were using those types interchangeably. Additionally, haddr_t documents the return value better than stating it as a size.